### PR TITLE
refactor: link both provisioners back to a shared node template

### PR DESCRIPTION
**Pull request by blacha**
#### Motivation

There appears to be a bug with cdk8s unsing nested specs causing volumes to show as 

```
volumeSize:
  value: 200Gi
```

the value key is invalid, doing the exact same source as a node template outpus

```
volumeSize: 200Gi
```

to work around the problem use a shared node template.

#### Modification

Adds a shared aws node template for both spot provisionsers to reference.

#### Checklist

_If not applicable, provide explanation of why._

- [ ] Tests updated
- [ ] Docs updated
- [ ] Issue linked in Title

### CODE REVIEW

File: config/charts/karpenter.ts [hunk 1]
```diff
@@ -39,13 +64,13 @@ export class Karpenter extends Chart {
...
-    new Helm(this, 'karpenter', {
+    const karpenter = new Helm(this, 'karpenter', {
       chart: 'oci://public.ecr.aws/karpenter/karpenter',
       namespace: 'karpenter',
-      version: 'v0.31.0',
+      version: 'v0.31.1',
       values: {
         serviceAccount: {
-          create: false,
+          create: true,
```
Comment (blacha) [inline config/charts/karpenter.ts:73]:
```suggestion
          create: false,
```

File: config/charts/karpenter.ts [hunk 2]
```diff
@@ -97,14 +119,14 @@ export class KarpenterProvisioner extends Chart {
           { key: 'kubernetes.io/arch', operator: 'In', values: ['amd64'] },
           { key: 'karpenter.k8s.aws/instance-family', operator: 'In', values: ['c5', 'c6i', 'c6a'] },
         ],
-        limits: { resources: { cpu: ProvisionerSpecLimitsResources.fromString('20000m') } },
-        provider,
+        limits: { resources: { cpu: ProvisionerSpecLimitsResources.fromNumber(2000) } },
```
Comment (l0b0) [inline config/charts/karpenter.ts:129]: Is this change intentional?

File: config/charts/karpenter.ts [hunk 3]
```diff
@@ -18,10 +43,10 @@ export class Karpenter extends Chart {
     super(scope, id, applyDefaultLabels(props, 'karpenter', 'v0.31.0', 'karpenter', 'workflows'));
 
     // Deploying the CRD
-    new Helm(this, 'karpenter-crd', {
+    const crd = new Helm(this, 'karpenter-crd', {
       chart: 'oci://public.ecr.aws/karpenter/karpenter-crd',
       namespace: 'karpenter',
-      version: 'v0.31.0',
+      version: 'v0.31.1',
```
Comment (blacha) [inline config/charts/karpenter.ts:49]: This should be a constant

File: config/charts/karpenter.ts (hunk 2 above)
Comment (blacha) [reply config/charts/karpenter.ts:129]: Yup, the cluster needs more than 20 cpus, this allows 2000 cpus to be spun up. (also likely too small)

**Review by paulfouquet — APPROVED** [review]
